python/lsst/ts/externalscripts/maintel/serpent_walk.py

```diff
@@ -0,0 +1,339 @@
+                                 LATISS. If set to zero, the rotator is positioned so that the
+                                 y-axis (dispersion axis) is aligned with the parallactic
+                                 angle.
+
+                    PhysicalSky: This strategy allows users to select the **initial** position
+                                 of the rotator in terms of the physical rotator angle (in the
+                                 reference frame of the telescope). Note that the telescope
+                                 will resume tracking the sky rotation.
+
+                    Physical: Select a fixed position for the rotator in the reference frame of
+                              the telescope. Rotator will not track in this mode.
+                type: string
+                enum: ["Sky", "SkyAuto", "Parallactic", "PhysicalSky", "Physical"]
+                default: Physical
+            required:
+              - total_time
+              - az_grid
+              - el_grid
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config: `types.SimpleNamespace`
+            Configuration data. See `get_schema` for information about data
+            structure.
+        """
+        self.log.debug(
+            f"Setting up configuration: \n"
+            f"  total_time: {config.total_time}\n"
+            f"  track_for: {config.track_for}\n"
+            f"  rot_value: {config.rot_value}\n"
+            f"  rot_type: {config.rot_type}\n"
+        )
+
+        self.config = config
+
+        self.config.rot_type = getattr(RotType, self.config.rot_type)
+
+        if hasattr(self.config, "ignore"):
+            for comp in self.config.ignore:
+                if comp not in self.tcs.components_attr:
+                    self.log.warning(
+                        f"Component {comp} not in CSC Group. "
+                        f"Must be one of {self.tcs.components_attr}. Ignoring."
+                    )
+                else:
+                    self.log.debug(f"Ignoring component {comp}.")
+                    setattr(self.tcs.check, comp, False)
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script."""
+        metadata.duration = self.config.total_time
+
+    async def run(self):
+
+        azel_generator = self.azel_grid_by_time(
+            total_time=self.config.total_time,
+            az_grid=self.config.az_grid,
+            el_grid=self.config.el_grid,
+            el_cutoff=self.config.el_cutoff,
+        )
+
+        for i, az, el in azel_generator:
+            self.log.debug(f"Serpent Walk - Target {i} - Moving to {az} / {el}")
+            await self.slew_and_track(az, el, target_name=f"serpent_walk_{i}")
+
+    def azel_grid_by_time(self, total_time, az_grid, el_grid, el_cutoff):
```

Comment:
Consider doing the same thing here as I suggested in the other script. Convert this to an async generator with no arguments.